src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/Dpi/FormDpiTests.cs

```diff
@@ -0,0 +1,71 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Runtime.InteropServices;
+using Xunit;
+using static Interop;
+
+namespace System.Windows.Forms.Tests.Dpi
+{
+    public class FormDpiTests : IClassFixture<ThreadExceptionFixture>
+    {
+        private IntPtr TriggerDpiMessage(User32.WM MSG, Control control, int newDpi)
+        {
+            double factor = newDpi / DpiHelper.LogicalDpi;
+            var wParam = PARAM.FromLowHigh(newDpi, newDpi);
+            var suggestedRect = new RECT(0, 0, (int)Math.Round(control.Width * factor), (int)Math.Round(control.Height * factor));
+
+            // Initialize unmanged memory to hold the struct.
+            IntPtr rectPtr = Marshal.AllocHGlobal(Marshal.SizeOf(suggestedRect));
+            Marshal.StructureToPtr(suggestedRect, rectPtr, true);
+            User32.SendMessageW(control.Handle, MSG, wParam, rectPtr);
+
+            return rectPtr;
+        }
```

Comment:
We avoid using `Marshal` as it introduces unnecessary overhead. 
`SendMessageW` takes care of blittable structs:
https://github.com/dotnet/winforms/blob/9c75beabab4eb6be306ea63ef2d8396c4e185cc2/src/System.Windows.Forms.Primitives/src/Interop/User32/Interop.SendMessageW.cs#L89-L99

...all you you need to write is this:

```suggestion
            RECT suggestedRect = new(0, 0, (int)Math.Round(control.Width * factor), (int)Math.Round(control.Height * factor));
            User32.SendMessageW(control, message, wParam, ref suggestedRect);
            return suggestedRect;
        }
```

Reply:
Also do we need to check the result of SendMessage: _If an application processes this message, it should return zero_?

Reply:
I think return value check is an overhead in this test and in either case test fails. 
Constructed SendMessageW with MSDN suggested pattern. Did you test you test your suggestion?

Reply:
We've been removing `Marshal` from the codebase for the past few years.